From e12d36edaea23b4ac8541eadbe59c47eef590b96 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Tue, 10 Nov 2020 00:23:11 +0100
Subject: [PATCH edk2-test 1/1] uefi-sct/SctPkg: illegal dereference in
 CheckEbcProtocol()

REF: https://bugzilla.tianocore.org/show_bug.cgi?id=3065

If LocateProtocol() has failed, variable Ebc is not valid and we may not
dereference it.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 .../BlackBoxTest/EfiCompliantBBTestPlatform_uefi.c          | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Generic/EfiCompliant/BlackBoxTest/EfiCompliantBBTestPlatform_uefi.c b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Generic/EfiCompliant/BlackBoxTest/EfiCompliantBBTestPlatform_uefi.c
index be06d139d59e..052f45b71599 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Generic/EfiCompliant/BlackBoxTest/EfiCompliantBBTestPlatform_uefi.c
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Generic/EfiCompliant/BlackBoxTest/EfiCompliantBBTestPlatform_uefi.c
@@ -3162,12 +3162,6 @@ CheckEbcProtocol (
                         );
     if (!EFI_ERROR (Status) && (SctStriCmp (String, L"yes") == 0)) {
       AssertionType = EFI_TEST_ASSERTION_FAILED;
-      if (!GenTestConfigTitle (IniFile, &AssertionType, L"EBCSupport")) {
-        GenTestConfigContent (L"Ebc->CreateThunk", Ebc->CreateThunk != NULL);
-        GenTestConfigContent (L"Ebc->UnloadImage", Ebc->UnloadImage != NULL);
-        GenTestConfigContent (L"Ebc->RegisterICacheFlush", Ebc->RegisterICacheFlush != NULL);
-        GenTestConfigContent (L"Ebc->GetVersion ", Ebc->GetVersion != NULL);
-      }
     }
   }
 
-- 
2.28.0

